examples/sprite-drawing/src/main.rs

```diff
@@ -0,0 +1,154 @@
+#![no_std]
+#![no_main]
+
+use core::{ffi::c_void, ptr};
+use psp::{sys::*, vram_alloc::*, *};
+
+// Width and height must be the same and be a power of 2.
+const IMAGE_SIZE: usize = 128;
+static FERRIS: Align16<[u8; IMAGE_SIZE * IMAGE_SIZE * 4 as usize]> =
+    Align16(*include_bytes!("../ferris.bin"));
+static mut LIST: psp::Align16<[u32; 0x40000]> = psp::Align16([0; 0x40000]);
+
+#[repr(C, align(4))]
+#[derive(Default, Clone, Copy)]
+pub struct Vertex {
+    u: f32,
+    v: f32,
+    color: u32,
+    x: f32,
+    y: f32,
+    z: f32,
+}
+
+psp::module!("sprite-example", 1, 1);
+
+fn psp_main() {
+    unsafe { psp_innermain() }
+}
+
+#[allow(unreachable_code)]
+unsafe fn psp_innermain() {
```

Comment:
not everything in this function is unsafe. It might also be wise to try to make some safe wrappers.